File: api/src/main/java/com/api/app/emum/PAY005.java
package com.api.app.emum;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * PG 코드: pgTypeCode
 * */
@Getter
@AllArgsConstructor
public enum PAY005 implements CommonCode {
    INICIS("001", "이니시스", 1, "50", ""),
    NICE("002", "나이스", 2, "50", ""),
    ;

    private final String code;
    private final String codeName;
    private final int displaySequence;
    private final String referenceValue1;
    private final String referenceValue2;

    public static PAY005 findByCode(String code) {
        return CommonCodeUtil.findByCode(PAY005.class, code);
    }
}

File: api/src/main/kotlin/com/api/app/emum/PAY005.kt
package com.api.app.emum

/** PG사코드: pgTypeCode */
enum class PAY005(
    override val code: String,
    override val codeName: String,
    override val displaySequence: Int,
    override val referenceValue1: String,
    override val referenceValue2: String
) : CommonCode {
    INICIS("001", "이니시스", 1, "0", ""),
    NICE("002", "나이스", 2, "0", ""),
    TEST("999", "테스트PG", 999, "100", ""),
    ;

    fun getReferenceValue1AsInt(): Int = referenceValue1.toInt()

    companion object {
        fun findByCode(code: String): PAY005? = CommonCodeUtil.findByCode(PAY005::class.java, code)
    }
}
